### PR TITLE
Moved architecture to docs/ folder to include it to documentation site

### PR DESCRIPTION
**Pull request by oliver-reload**
### Link to issue
[Please add a link to the issue being addressed by this change.](https://reload.atlassian.net/browse/DDFMINI-10)

### Description

Our new documentation site built on mkdocs builds our documentation site automagically by looking at the structure in /docs (default). I am consolidating the standard in our DPL repos to accomodate this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,734 +6,5 @@
...
-    - [Creating an atom](#creating-an-atom)
-    - [Creating a component](#creating-a-component)
-  - [Editor example configuration](#editor-example-configuration)
-- [Usage](#usage)
-  - [Naive app mount](#naive-app-mount)
-    - [Data attributes and props](#data-attributes-and-props)
-- [Extending the project](#extending-the-project)
-
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
-<!-- markdownlint-enable -->
```
Comment (kasperg) [inline README.md:39]: Remember to remove doctoc.

**Review by kasperg — CHANGES_REQUESTED** [review]

File: docs/ui_text_handling.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The main purpose of the functionality is to be able to access strings defined
 at app level inside of sub components
 without passing them all the way down via props.
 You can read more about the decision
-and considerations [here](../architecture/adr-002-ui-text-handling.md).
+and considerations [here](../docs/architecture/adr-002-ui-text-handling.md).
```
Comment (kasperg) [inline docs/ui_text_handling.md:12]: Should this just be `./architecture/adr-002-ui-text-handling.md`?

**Review by kasperg — APPROVED** [review]
:+1: LGTM. One change for you to consider before this is merged.

File: docs/ui_text_handling.md (hunk 2 above)
Comment (oliver-reload) [reply docs/ui_text_handling.md]: Done. 
- And keep in mind someone probably needs to go through and change most relative links anyway, as described on the documentation site.

**Pull request merged**